tests/integration/api/test_project.py

```diff
@@ -0,0 +1,151 @@
+    }
+    response = await client.post(
+        "/api/new-project/",
+        json=bad_project,
+    )
+    assert response.status_code == status_codes.HTTP_422_UNPROCESSABLE_ENTITY
+
+
+@pytest.mark.asyncio
+async def test_get_project(client: AsyncClient, db: AsyncSession):
+
+    query = sa.insert(models.projects).values(
+        url=TEST_PROJECT["url"],
+        name=TEST_PROJECT["name"],
+        description=TEST_PROJECT["description"],
+    )
+
+    await db.execute(query)
+
+    projects = (await db.execute(sa.select(models.projects))).all()
+    assert len(projects) == 1
+    project = projects[0]
+
+    response = await client.get("/api/project/" + str(project["id"]))
+
+    assert response.status_code == status_codes.HTTP_200_OK
+
+    data = response.json()
+    assert data["name"] == TEST_PROJECT["name"]
+    assert data["url"] == TEST_PROJECT["url"]
+
+
+@pytest.mark.asyncio
+async def test_get_project_not_found(client: AsyncClient, db: AsyncSession):
+
+    query = sa.insert(models.projects).values(
+        url=TEST_PROJECT["url"],
+        name=TEST_PROJECT["name"],
+        description=TEST_PROJECT["description"],
+    )
+
+    await db.execute(query)
+
+    projects = (await db.execute(sa.select(models.projects))).all()
+    assert len(projects) == 1
+    project = projects[0]
+
+    response = await client.get("/api/project/4")
+
+    assert response.status_code == status_codes.HTTP_404_NOT_FOUND
+
+
+@pytest.mark.asyncio
+async def test_get_projects(client: AsyncClient, db: AsyncSession):
+
+    test_project_two = {
+        "url": "https://github.com/benthomasson/eda-project",
+        "name": "Test Name TWO",
+        "description": "This is a test description two",
+    }
+
+    test_projects = [TEST_PROJECT, test_project_two]
+
+    for test_project in test_projects:
+        query = sa.insert(models.projects).values(
+            url=test_project["url"],
+            name=test_project["name"],
+            description=test_project["description"],
+        )
+        await db.execute(query)
+
+    projects = (await db.execute(sa.select(models.projects))).all()
+    assert len(projects) == 2
+
+    response = await client.get("/api/projects/")
+
+    assert response.status_code == status_codes.HTTP_200_OK
+
+    data = response.json()
+    assert len(data) == 2
+    assert data[0]["name"] == TEST_PROJECT["name"]
+    assert data[1]["url"] == test_project_two["url"]
+
+
+@pytest.mark.asyncio
+async def test_edit_project(client: AsyncClient, db: AsyncSession):
+
+    query = sa.insert(models.projects).values(
+        url=TEST_PROJECT["url"],
+        name=TEST_PROJECT["name"],
+        description=TEST_PROJECT["description"],
+    )
+
+    await db.execute(query)
+
+    projects = (await db.execute(sa.select(models.projects))).all()
+    assert len(projects) == 1
+    project = projects[0]
+
+    response = await client.patch(
+        "/api/project/" + str(project["id"]) + "/edit",
```

Comment:
Prefer string formatting to concatenation.

```suggestion
        f"/api/project/{project["id"]}/edit",
```